src/routes/search.js

```diff
@@ -150,6 +151,11 @@ class SearchRoute extends Component {
     // if the new query is different from our location, update the location
     if (qs.stringify(params) !== qs.stringify(this.state.params)) {
       let url = this.buildSearchUrl(location.pathname, params);
+
+      // if new query is different from current query this will close detail pane
+      if (params.q !== this.state.params.q) {
+        url = this.buildSearchUrl('/eholdings', params);
+      }
```

Comment:
🎵 _Can it be that it was all so simple then?_ 🎵

Reply:
...But we rather make it hard. But writing code is like a battle and we all end up with scars...